### PR TITLE
Add cover & menu thumbnail images deleting for category

### PR DESCRIPTION
**Pull request by sarjon**
Status: ready to merge after rebase

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Allows deleting cover and menu thumbnail images when editing category.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/10194
| How to test?  | You should be able to delete images from category edit page as you were able in legacy page. admin-dev/sell/catalog/categories

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11481)

### CODE REVIEW

File: src/Core/Domain/Category/ValueObject/CategoryId.php
Comment (mickaelandrieu) [inline src/Core/Domain/Category/ValueObject/CategoryId.php]: you shouldn't do that here, instead better validate the input data :+1:

File: src/Core/Domain/Category/ValueObject/MenuThumbnailId.php
Comment (mickaelandrieu) [inline src/Core/Domain/Category/ValueObject/MenuThumbnailId.php]: same here, avoid casting

Comment (sarjon) [reply src/Core/Domain/Category/ValueObject/MenuThumbnailId.php]: problem is that data that comes from forms or query params are always string, the only solution i can think of is to cast id before passing to constructor, wdyt?

Comment (mickaelandrieu) [reply src/Core/Domain/Category/ValueObject/MenuThumbnailId.php]: I think it's better even if it's ugly right now :/

 Remember what we are doing with Filters in Grids, I think "in the end" we may imagine to "clean" the data that comes from the request before entering in the controller.

Or work at Form level (typical use case: getNormData() and ensure that we retrieve an integer when we expect an integer)

If we use an IntegerType field, getNormData() must return an integer.


> we can also implement our own Data transformers to ensure the behavior

File: admin-dev/themes/new-theme/js/components/form-submit-button.js
Comment (matks) [inline admin-dev/themes/new-theme/js/components/form-submit-button.js:51]: 👍 nice one !

Comment (matks) [reply admin-dev/themes/new-theme/js/components/form-submit-button.js:51]: Does this mean we should go back on already migrated pages to use this component everywhere ?

File: src/Core/Domain/Category/ValueObject/MenuThumbnailId.php
Comment (matks) [reply src/Core/Domain/Category/ValueObject/MenuThumbnailId.php]: 👍

**Review by matks — DISMISSED** [review]
No changes except the comment from Mickael 👍

File: admin-dev/themes/new-theme/js/components/form-submit-button.js
Comment (sarjon) [reply admin-dev/themes/new-theme/js/components/form-submit-button.js:51]: we could, but its not must. :)

**Comment by sarjon** [comment]
@matks done. :)

File: src/Adapter/Category/CommandHandler/AddCategoryHandler.php
Comment (PierreRambaud) [inline src/Adapter/Category/CommandHandler/AddCategoryHandler.php:62]: Why not casting in directly in CategoryId?

Comment (sarjon) [reply src/Adapter/Category/CommandHandler/AddCategoryHandler.php:62]: see comments above. as i understand, it only validates that provided value is actual `int`, but does not fix it for you, as it may be bug.

**Comment by sarjon** [comment]
@matks i think i've addressed all the comments.

**Review by matks — DISMISSED** [review]

**Comment by marionf** [comment]
@ntiepresta be careful actions "Add new root category", "Add new category", "Edit" and categories listing PR #10408 has not been tested, so we should also check that

**Comment by marionf** [comment]
@ntiepresta be careful, previous PR has not been tested, so we need to test the whole add/edit category page

**Comment by sarjon** [comment]
@marionf note that not all functionality of Category form is migrated yet, so its pretty hard to test. :/

**Comment by marionf** [comment]
All right @sarjon, so we will just test deleting cover and menu thumbnail images

**Comment by sarjon** [comment]
thanks @marionf, either way we will have to test it a lot after it is finished since its pretty complicated form. :/

**Comment by marionf** [comment]
It's ok for the deletion of category cover image, category thumbnail and menu thumbnails

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Rebased, waiting for Travis before merge...

**Comment by sarjon** [comment]
@matks regarding this error:

_Caused by
GuzzleHttp\Exception\ConnectException: cURL error 28: Operation timed out after 5001 milliseconds with 52198 bytes received_

if i understand correctly, it does actual http request? shouldnt it be mocked it tests instread? :thinking:

**Comment by PierreRambaud** [comment]
It is a server timeout, i rerun tests for you @sarjon

**Comment by PierreRambaud** [comment]
Still waiting for wording before merge?

**Comment by sarjon** [comment]
Wording was confirmed already, its just @prestonBot who enjoys adding labels. -_-

**Comment by matks** [comment]
Thanks @sarjon !

**Comment by matks** [comment]
Thanks @PierreRambaud !

**Pull request merged**
